src/iconv.jl

```diff
@@ -1,9 +1,55 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
 module iconv
-import Base: close, eof, flush, read, readall, write
+import Base: close, eof, flush, read, readall, write, show
 import Base.Libc: errno, strerror, E2BIG, EINVAL, EILSEQ
 export StringEncoder, StringDecoder, encode, decode
+export StringEncodingError, OutputBufferError, IConvError
+export InvalidEncodingError, InvalidSequenceError, IncompleteSequenceError
+
+
+abstract StringEncodingError
+
+# Specified encodings or the combination are not supported by iconv
+type InvalidEncodingError <: StringEncodingError
+    args::Tuple{ASCIIString, ASCIIString}
+end
+InvalidEncodingError(from, to) = InvalidEncodingError((from, to))
+message(::Type{InvalidEncodingError}) = "Conversion from <<1>> to <<2>> not supported by iconv implementation, check that specified encodings are correct"
+
+# Encountered invalid byte sequence
+type InvalidSequenceError <: StringEncodingError
+    args::Tuple{ASCIIString}
+end
+InvalidSequenceError(seq) = InvalidSequenceError((bytes2hex(seq),))
+message(::Type{InvalidSequenceError}) = "Byte sequence 0x<<1>> is invalid in source encoding or cannot be represented in target encoding"
+
+type IConvError <: StringEncodingError
+    args::Tuple{ASCIIString, Int, ASCIIString}
+end
+IConvError(func) = IConvError((func, errno(), strerror(errno())))
+message(::Type{IConvError}) = "<<1>>: <<2>> (<<3>>)"
+
+# Input ended with incomplete byte sequence
+type IncompleteSequenceError <: StringEncodingError ; end
+message(::Type{IncompleteSequenceError}) = "Incomplete byte sequence at end of input"
+
+type OutputBufferError <: StringEncodingError ; end
+message(::Type{OutputBufferError}) = "Ran out of space in the output buffer"
+
+type InvalidBytesError <: StringEncodingError ; end
```

Comment:
Ah, I see. Then just remove that branch, as I don't think it can happen (the call doesn't pass any input data...). Anyway we'll get a generic error if it happens.